src/broker/controller-dbus.c

```diff
@@ -528,6 +528,26 @@ static int controller_dispatch_reply(Controller *controller, uint32_t serial, co
         return 0;
 }
 
+static int controller_dispatch_error(Controller *controller, uint32_t serial, const char *error_name, Message *message) {
+        ControllerReload *reload;
+        int r;
+
+        reload = controller_find_reload(controller, serial);
+        if (!reload)
+                return CONTROLLER_E_UNEXPECTED_REPLY;
+
+        if (strcmp(error_name, "org.bus1.DBus.Controller.Error.InvalidConfig"))
+                return CONTROLLER_E_UNEXPECTED_ERROR;
```

Comment:
Errr, shouldn't we invalidate the reload regardless of the error? I mean, we got a reply that matches the serial of our request, it seems counter-intuitive to ignore them?

Reply:
On the other hand, getting an unknown error from our own controller seems like an error we should catch. But if we ever want to open up the interface to external controllers, I think we should catch all errors here and treat them as failed reload, right?

Nothing critical, though.

Reply:
Hm, I'd say we should consider error codes API and not allow people to add random ones. But let's burn that bridge when we get to it.